be/src/olap/delta_writer_v2.cpp

```diff
@@ -125,6 +125,9 @@ Status DeltaWriterV2::init() {
     context.partial_update_info = _partial_update_info;
     context.memtable_on_sink_support_index_v2 = true;
 
+    auto tablet = DORIS_TRY(ExecEnv::GetInstance()->storage_engine().get_tablet(_req.tablet_id));
```

Comment:
DeltaWriterV2 is on the upstream BE, the tablet requested here can be on downstream BE.
This causes the following problem:

```
I20241110 23:46:32.273124  5731 pipeline_fragment_context.cpp:170] PipelineFragmentContext::cancel|query_id=4334341c4795459e-ade78c01be07735d|fragment_id=0|reason=[INTERNAL_ERROR]failed to get tablet: 14168, reason: tablet does not exist. 172.20.50.97
        0#  doris::StorageEngine::get_tablet(long) at /home/zcp/repo_center/doris_master/doris/be/src/common/status.h:375
        1#  doris::DeltaWriterV2::init() at /home/zcp/repo_center/doris_master/doris/be/src/util/expected.hpp:1986
        2#  doris::DeltaWriterV2::write(doris::vectorized::Block const*, std::vector<unsigned int, std::allocator<unsigned int> > const&) at /home/zcp/repo_center/doris_master/doris/be/src/common/status.h:499
        3#  doris::vectorized::VTabletWriterV2::_write_memtable(std::shared_ptr<doris::vectorized::Block>, long, doris::vectorized::Rows const&) at /home/zcp/repo_center/doris_master/doris/be/src/common/status.h:388
        4#  doris::vectorized::VTabletWriterV2::write(doris::RuntimeState*, doris::vectorized::Block&) at /var/local/ldb-toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/shared_ptr_base.h:701
        5#  doris::vectorized::AsyncResultWriter::process_block(doris::RuntimeState*, doris::RuntimeProfile*) at /home/zcp/repo_center/doris_master/doris/be/src/common/status.h:499
        6#  std::_Function_handler<void (), doris::vectorized::AsyncResultWriter::start_writer(doris::RuntimeState*, doris::RuntimeProfile*)::$_0>::_M_invoke(std::_Any_data const&) at /var/local/ldb-toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/ext/atomicity.h:98
        7#  doris::ThreadPool::dispatch_thread() at /home/zcp/repo_center/doris_master/doris/be/src/util/threadpool.cpp:0
        8#  doris::Thread::supervise_thread(void*) at /var/local/ldb-toolchain/bin/../usr/include/pthread.h:562
        9#  ?
        10# ?
```

Reply:
I meet this exception when I insert rows into olap table when the tablet not exist in backend of the instance:


```sql
CREATE TABLE `a4` (
  `date_time` date NULL
) ENGINE=OLAP
DUPLICATE KEY(`date_time`)
PARTITION BY RANGE(`date_time`)
(PARTITION p1 VALUES [('0000-01-01'), ('2020-01-02')),
PARTITION p2 VALUES [('2020-01-02'), ('2020-01-03')),
PARTITION p3 VALUES [('2020-01-03'), ('2020-01-04')))
DISTRIBUTED BY HASH(`date_time`) BUCKETS 1
PROPERTIES (
"replication_allocation" = "tag.location.default: 1",
"min_load_replica_num" = "-1",
"is_being_synced" = "false",
"storage_medium" = "hdd",
"storage_format" = "V2",
"inverted_index_storage_format" = "V2",
"light_schema_change" = "true",
"disable_auto_compaction" = "false",
"enable_single_replica_compaction" = "false",
"group_commit_interval_ms" = "10000",
"group_commit_data_bytes" = "134217728"
);

insert into a4 values('2020-01-03');
insert into a4 values('2020-01-03');
insert into a4 values('2020-01-03');

(1105, 'errCode = 2, detailMessage = (192.168.126.3)[INTERNAL_ERROR]failed to get tablet: 180160, reason: tablet does not exist. 192.168.126.3')
```
for the above sql, I have 2 backends in doris, and create `a4` with one tablet, the insert statement will select a random backend to execute. when the backend of the instance not contains the tablets of `a4`,  you will meet the exception